### PR TITLE
WFCORE-994 Make cross process wildcard r-r-d work

### PR DESCRIPTION
**Pull request by kabir**
https://issues.jboss.org/browse/WFCORE-994
https://issues.jboss.org/browse/WFCORE-948 (which is similar to https://issues.jboss.org/browse/WFCORE-315, which also gets solved)
https://issues.jboss.org/browse/WFCORE-1027

### CODE REVIEW

**Comment by kabir** [comment]
Labelling 'hold' to give @hpehl and @heiko-braun a chance to try it out first.

**Comment by heiko-braun** [comment]
tnx, @kabir. i'll have a look today or tomorrow

**Comment by kabir** [comment]
Retest this please

File: controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java [hunk 1]
```diff
@@ -772,55 +815,47 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
...
 
-                final ModelNode result = this.result.add();
-                result.get(OP_ADDR).set(base.toModelNode());
-                context.addStep(result, newOp, handler, OperationContext.Stage.MODEL, true);
+            boolean isServerWildcardWithRemaining = currentElement.getKey().equals(RUNNING_SERVER) && currentElement.isWildcard() && newRemaining.size() > 0;
+            for (final PathElement path : children) {
+                if (childType != null && !childType.equals(path.getKey())) {
+                    continue;
+                }
+                if (isServerWildcardWithRemaining && path.getKey().equals(RUNNING_SERVER) && path.isWildcard() && newRemaining.size() > 0) {
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java]: This confuses me. L837 is doing a similar check on "currentElement" as is being done here on "path". And the newRemaining.size() > 0 check is redundant.

**Comment by bstansberry** [comment]
Other than my one question, looks good.

**Comment by heiko-braun** [comment]
See my comment son the following example queries:

- server resources: https://gist.github.com/heiko-braun/695790e82ac35704f4b1
- host resources: https://gist.github.com/heiko-braun/416feaa6726483684be2
- server-config resources: https://gist.github.com/heiko-braun/e73cbe424199409fde37

**Comment by heiko-braun** [comment]
The AFTER situation:

- host resources: https://gist.github.com/heiko-braun/78b6d9b94971283228ec
- server resources: https://gist.github.com/heiko-braun/f1e3e81635ad2cc72da9
- server-config resources: https://gist.github.com/heiko-braun/daacba231778955e1946

However with privileges enabled it runs into an exception:

- https://gist.github.com/heiko-braun/8e152c250c0bd56daea3
- https://gist.github.com/heiko-braun/99ed3067d424d0cbde89

**Comment by heiko-braun** [comment]
Here's the RBAC config for the setup that fails:

- https://gist.github.com/heiko-braun/0dc810ed04db8739defd

**Comment by kabir** [comment]
Retest this please

**Comment by heiko-braun** [comment]
If these exceptiondon't appear anymore we are good to go:

However with privileges enabled it runs into an exception:

https://gist.github.com/heiko-braun/8e152c250c0bd56daea3
https://gist.github.com/heiko-braun/99ed3067d424d0cbde89

**Comment by kabir** [comment]
Retest this please

**Pull request merged**